src/common/common.h

```diff
@@ -34,10 +56,11 @@
     backtrace_symbols_fd(buffer, calls, 1);                               \
     exit(-1);                                                             \
   } while (0);
+#endif
 
-#define CHECKM(COND, M, ...)                                 \
-  if (!(COND)) {                                             \
-    LOG_ERR("Check failure: %s \n" M, #COND, ##__VA_ARGS__); \
+#define CHECKM(COND, M, ...)                                   \
+  if (!(COND)) {                                               \
+    LOG_ERROR("Check failure: %s \n" M, #COND, ##__VA_ARGS__); \
```

Comment:
`CHECKM` definitely should be fatal.

That said, do we actually want LOG_WARN, LOG_ERROR, and LOG_FATAL? It seems like either we can recover, in which case it should be LOG_WARN, or we can't, in which case it should be LOG_FATAL.

I think pretty much all of the LOG_ERRORs that we currently have should potentially be fatal errors.